yt_dlp/extractor/acfun.py

```diff
@@ -0,0 +1,162 @@
+import re
+
+from .common import InfoExtractor
+from ..utils import (
+    determine_ext,
+    float_or_none,
+    int_or_none,
+    traverse_obj,
+)
+
+
+class AcFunVideoIE(InfoExtractor):
+    IE_NAME = 'AcFunVideoIE'
+    _VALID_URL = r'(?x)https?://(?:www\.acfun\.cn/v/)ac(?P<id>[_\d]+)'
+
+    _TESTS = [{
+        'url': 'https://www.acfun.cn/v/ac35457073',
+        'info_dict': {
+            'id': '35283657',
+            'title': '【十五周年庆】文章区UP主祝AcFun生日快乐！',
+            'duration': 455.21,
+            'timestamp': 1655289827,
+        },
+        'params': {
+            # m3u8 download
+            'skip_download': True,
+        },
+        'skip': 'Geo-restricted to China',
+    }, {
+        'url': 'https://www.acfun.cn/v/ac35457073',
+        'only_matching': True,
+    }]
+
+    @staticmethod
+    def parse_format(info):
+        vcodec, acodec = None, None
+
+        codec_str = info.get('codecs') or ''
+        m = re.match('(?P<vc>[^,]+),(?P<ac>[^,]+)', codec_str)
+        if m:
+            vcodec = m.group("vc")
+            acodec = m.group("ac")
+
+        return {
+            'url': info.get('url'),
+            'fps': int_or_none(info.get('frameRate')),
+            'width': int_or_none(info.get('width')),
+            'height': int_or_none(info.get('height')),
+            'vcodec': vcodec,
+            'acodec': acodec,
+            'tbr': float_or_none(info.get('avgBitrate'))
+        }
+
+    def _real_extract(self, url):
+        mobj = self._match_valid_url(url)
+        video_id = mobj.group('id')
+
+        webpage = self._download_webpage(url, video_id)
+        json_all = self._search_json(r'window.videoInfo\s*=\s*', webpage, 'videoInfo', video_id)
+
+        video_info = json_all['currentVideoInfo']
+        playjson = self._parse_json(video_info['ksPlayJson'], video_id)
+        video_inner_id = traverse_obj(json_all, ('currentVideoInfo', 'id'))
+
+        ext = determine_ext(video_info.get('fileName'), 'mp4')
+
+        formats = []
+        for video in traverse_obj(playjson, ('adaptationSet', 0, 'representation')):
+            fmt = AcFunVideoIE.parse_format(video)
+            formats.append({**fmt, 'ext': ext})
```

Comment:
This lines seems can generalized

Reply:
Sorry for bad wording, you can set this as base class ( which i preffered) or as class method to avoid writing same coding multiple times